compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs

```diff
@@ -3293,6 +3293,14 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
                 let mut parent_trait_pred =
                     self.resolve_vars_if_possible(data.derived.parent_trait_pred);
                 let parent_def_id = parent_trait_pred.def_id();
+                if tcx.is_diagnostic_item(sym::FromResidual, parent_def_id)
+                    && !tcx.features().enabled(sym::try_trait_v2)
```

Comment:
We could add a further check for whether the current compiler is nightly and the feature is disabled to still provide a note that the feature could be enabled.